cou/steps/plan.py

```diff
@@ -88,7 +90,7 @@ async def create_upgrade_group(
     target: str,
     description: str,
     filter_function: Callable[[OpenStackApplication], bool],
-) -> UpgradeStep:
+) -> Optional[UpgradeStep]:
```

Comment:
I do not see how this can returns `None`. I think it will always returns `group_upgrade_plan`.